chia/consensus/blockchain.py

```diff
@@ -788,12 +788,16 @@ async def validate_unfinished_block(
 
         return PreValidationResult(None, required_iters, conds, uint32(0))
 
-    def contains_block(self, header_hash: bytes32) -> bool:
-        """
-        True if we have already added this block to the chain. This may return false for orphan blocks
-        that we have added but no longer keep in memory.
-        """
-        return header_hash in self.__block_records
+    def contains_block(self, header_hash: bytes32, height: Optional[uint32] = None) -> bool:
+        if height is None:
+            block_rec = self.try_block_record(header_hash)
+            if block_rec is None:
+                return False
```

Comment:
this seems a bit risky, don't you think?
The caller think they're asking if the block is part of the main chain, but gets false if it's not in the cache, even if it is part of the main chain.

This should be clearly documented.